src/components/visit/visitList.tsx

```diff
@@ -0,0 +1,55 @@
+import { useEffect, useState } from 'react'
+import { getVisitData, Visit } from 'databaseIntigration'
+
+import VisitInstance from './visitInstance'
+
+interface VisitListProps {
+  searchQuery: string
+}
+
+export default function VisitList(props: VisitListProps) {
+  const [visitData, setVisitData] = useState<Visit[]>([])
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const response: Visit[] = await getVisitData()
+      setVisitData(response)
+    }
+    fetchData()
+  }, [])
+
+  return (
+    <div className=''>
+      {visitData
+        .filter((post: Visit) => {
+          if (
+            props.searchQuery === '' ||
+            post.firstName
+              .toLowerCase()
+              .includes(props.searchQuery.toLowerCase()) ||
+            post.lastName
+              .toLowerCase()
+              .includes(props.searchQuery.toLowerCase()) ||
+            post.petName.toLowerCase().includes(props.searchQuery.toLowerCase())
+          ) {
+            return post
```

Comment:
Whenever you have a lot of complex boolean logic like this I find it is much nicer to extract it out into some well named functions.

Consider:

```tsx
const matchesFirstName = (post: Visit) =>
    post.firstName
    .toLowerCase()
    .includes(props.searchQuery.toLowerCase());

const matchesLastName = (post: Visit) =>
    post.lastName
    .toLowerCase()
    .includes(props.searchQuery.toLowerCase());

const matchesSearchTerms = (post: Visit) => matchesFirstName(post) || matchesLastName(post);
```

and then when you filter:

```tsx
visitData
  .filter(matchesSearchTerms)
  .map(....)
```